docs/client-sdk/javascript/tutorials/build-an-xmtp-hello-world-app.md

```diff
@@ -0,0 +1,576 @@
+     autoConnect: true,
+     connectors: modalConnectors({ appName: "web3Modal", chains }),
+     provider,
+   });
+   // Web3Modal Ethereum Client
+   const ethereumClient = new EthereumClient(wagmiClient, chains);
+   // highlight-start
+   type MessageListProps = {
+     msg: DecodedMessage[];
+   };
+   // highlight-end
+
+   function App() {
+     const { isConnected } = useAccount();
+     // highlight-start
+     const [messages, setMessages] = useState<DecodedMessage[]>([]);
+     const [client, setClient] = useState<any>();
+     const [xmtpClientAddress, setXmtpClientAddress] = useState<any>();
+     // highlight-end
+
+     // highlight-start
+     const initXmtp = async function () {
+       const signer = await fetchSigner();
+       const xmtp = await Client.create(signer, { env: "production" });
+       const conversation = await xmtp.conversations.newConversation(PEER_ADDRESS);
+       const messages = await conversation.messages({
+         direction: SortDirection.SORT_DIRECTION_DESCENDING,
+       });
+
+       setClient(conversation);
+       setMessages(messages);
+       setXmtpClientAddress(xmtp.address);
+       // highlight-end
+     };
+     // highlight-start
+     useEffect(() => {
+       if (xmtpClientAddress) {
+         const streamMessages = async () => {
+           const newStream = await client.streamMessages();
+           for await (const msg of newStream) {
+             setMessages((prevMessages) => {
+               const messages = [...prevMessages];
+               messages.unshift(msg);
+               return messages;
+             });
+           }
+         };
+         streamMessages();
+       }
+     }, [client, xmtpClientAddress]);
+     // highlight-end
+
+     // highlight-start
+     const onSendMessage = async () => {
+       const message = "gm XMTP bot!";
+       await client.send(message);
+     };
+    // highlight-end
+
+     // highlight-start
+     const MessageList: React.FC<MessageListProps> = ({ msg }) => {
+       return (
+         <ul>
+           {msg.map((message, index) => (
+             <li key={index}>{message.content}</li>
+           ))}
+         </ul>
+       );
+     };
+     // highlight-end
+
+     return (
+       <div className="App">
+         <Web3Button />
+         <Web3Modal
+           projectId="<YOUR_PROJECT_ID>"
+           ethereumClient={ethereumClient}
+         />
+         // highlight-start
+         {isConnected && xmtpClientAddress && (
+           <>
+             <MessageList msg={messages} />
+             <button onClick={onSendMessage}>Send GM</button>
+           </>
+         )}
+         {isConnected && !xmtpClientAddress && (
+           <button onClick={initXmtp}>Connect to XMTP</button>
+         )}
+         // highlight-end
+       </div>
+     );
+   }
+
+   export default App;
+
+   ```
+   When you run the code above, it should give you a blank skeleton like the picture below:
+   ![barebone bot app](img/gm-bot-bare.png)
```

Comment:
I am able to connect my wallet and connect to XMTP! 

When I first connected, I didn't see all of the messages I've exchanged with the bot in the past -- it seemed to be only displaying the last message I sent and received. I disconnected and then connected again and same. I sent a new message and it showed up in XMTP GM app, but it didn't show up in xmtp.chat. 

I worked on some other stuff and about 10 minutes later, I reloaded the app and all of the messages are showing up! YAY! I sent a message and it showed up in XMTP GM app and xmtp.chat! However, the one I sent 10 minutes earlier isn't showing up - I wonder where that message went?

I wonder if it is just a glitch with me - or if others are seeing it?

Reply:
We discussed via slack. Looks like it was a glitch first time around. Resolving for now